### PR TITLE
[MU4] Update to latest SMuFL version 1.3

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
As taken from https://github.com/w3c/smufl

Counterpart #6543 is in 3.x now

### CODE REVIEW

**Comment by lasconic** [comment]
Maybe sym.cpp and sym.h need to be updated too ?

**Comment by Jojo-Schmitz** [comment]
Hmm, true, very good point, there are a few new glyphs. Only not even Bravura contains them yet as far as I can tell

**Comment by Jojo-Schmitz** [comment]
Does it matter where exactly they get added to sym.{cpp,h}?

**Comment by Jojo-Schmitz** [comment]
Ah, looks like they are in strict alphabethical order in sym.{cpp,h}
Problem: `ornamentMordentInverted` got removed, but is used outside sym.{cpp,h}

**Comment by Jojo-Schmitz** [comment]
Actually `ornamentMordentInverted` turned into `ornamentMordent` that into `ornamentShortTrill`, codepoint-wise

**Comment by Jojo-Schmitz** [comment]
Didn't you have some (Python?) script to update those files automagically?

**Comment by Jojo-Schmitz** [comment]
Done, but I'm not sure whether I got the `ornamentMordentInverted` vs. `ornamentMordent` and `ornamentShortTrill` right in all cases

**Comment by Jojo-Schmitz** [comment]
Something is bad, need to check again, seems somehow the order of the glyphs got messed up

**Comment by lasconic** [comment]
I used this script in the past : https://gist.github.com/lasconic/be08234b3dd074e99edc1fb060f0abcb

**Comment by Jojo-Schmitz** [comment]
Workspace needs some work still, the Ornaments palette needs updating

**Comment by Jojo-Schmitz** [comment]
@lasconic could you run your script to verify my 'handmade' changes?

**Comment by Jojo-Schmitz** [comment]
as far as I can tell the Travis CO mtests failures are all due to the mordent changes. These definitly need a good review before looking at how to fix those mtest failures

**Comment by mirabilos** [comment]
Please also update the Unix manpage https://github.com/musescore/MuseScore/blob/master/build/Linux%2BBSD/mscore.1.in#L511 in the same commit, so the data stays consistent.

Thanks!

**Comment by Jojo-Schmitz** [comment]
done

**Comment by RobFog** [comment]
Might make sense to document somewhere that updating the manpage should go along with an SMuFL update.

**Comment by Jojo-Schmitz** [comment]
There's no suitable place for that I'm affraid

**Comment by RobFog** [comment]
The Developer Handbook would seem like a good place.

@mirabilos Are there other components where Unix manpage updates are also necessary?

**Comment by mirabilos** [comment]
Robbie dixit:

>@mirabilos Are there other components where Unix manpage updates are
>@also necessary?

The hyperlinks in the SEE ALSO section (immediately above the
STANDARDS section in which the SMuFL version is noted), and in
the STANDARDS section, MusicXML, SoundFont and SFZ.

I did a quick pass through the manpage, and I think these are it.

Command line option changes (in mscore/musescore.cpp mostly)
and changes to the portable application mechanism need to be
also reflected.

**Comment by Jojo-Schmitz** [comment]
I meant no good place in the code, so nothing for this PR

**Comment by RobFog** [comment]
Thanks, @mirabilos.

> I meant no good place in the code, so nothing for this PR
Ah, yes.

@Jojo-Schmitz, do you think this should go into the Handbook?

**Comment by anatoly-os** [comment]
@Jojo-Schmitz what is the reason to update the font? Some known issues with current version? User requests?

**Comment by Jojo-Schmitz** [comment]
Just to stay up-to-date and on par with Steinberg/Dorico ;-)

**Comment by anatoly-os** [comment]
So, there were no requests or known bugs, right?

**Comment by Jojo-Schmitz** [comment]
None for MuseScore, but apparently some for SMuFL, otherwise there won't be changes there?

**Comment by MarcSabatella** [comment]
In general, I certainly favor keeping up to date with the latest version of this (or any) font, as presumably there are improvements we would want to take advantage of.  But we'd also want to make sure there aren't incompatible changes that could bite us.  Didn't this happen recently-ish with, hmm, position of repeat dots or something like that?

**Comment by Jojo-Schmitz** [comment]
That's why I'd like it in now, with a proper Beta phase

**Comment by Jojo-Schmitz** [comment]
As per https://github.com/w3c/smufl/milestone/3 SMuFL 1.4 is due by the end of this year

**Comment by Jojo-Schmitz** [comment]
As per https://github.com/w3c/smufl/milestone/3 SMuFL 1.4 is due by the end of this year. But apparently no work on it has yet started...

**Comment by dmitrio95** [comment]
@Jojo-Schmitz could you please force-push changes here to trigger vtest (see my comment in https://github.com/musescore/MuseScore/pull/4879#issuecomment-598080805)?

**Comment by Jojo-Schmitz** [comment]
done

**Comment by AntonioBL** [comment]
There is a change in musejazz-4, see https://github.com/musescore/MuseScore/suites/516146421/artifacts/2824761
Unfortunately, the bot for commenting the PR didn't work 😞 
Maybe adding a log warning could be the best option.

**Comment by Jojo-Schmitz** [comment]
seen that but couldn't make any sense of it

**Comment by Jojo-Schmitz** [comment]
So yes, something to fix in this PR I guess

**Comment by Jojo-Schmitz** [comment]
Not sure what/how though :-(

**Comment by Jojo-Schmitz** [comment]
OK, the difference is that the vtest scores for  Emmentaler, Bravura and Gonville are MuseScore 2 score, and so run thru `read206()`, while the museJazz one is a MuseScore 3 score. I could fix that musejazz-4.mscx by  turning `ornamentMordentInverted` into `ornamentMordent` and that into `ornamentShortTrill`, but that would rather be curing the symptoms, wouldn't it?

**Comment by Jojo-Schmitz** [comment]
We'd have a down rev problem too, as older versions of MuseScore won't understand the new `ornamentShortTrill` and not render it at all.

**Comment by Jojo-Schmitz** [comment]
Submitted an issue for SMuFL: https://github.com/w3c/smufl/issues/119

**Comment by dmitrio95** [comment]
Let's wait for a reply in https://github.com/w3c/smufl/issues/119 then prior to merging this PR.

**Comment by Jojo-Schmitz** [comment]
Yes, I'm afraid we'd have to. I should have raised that issue there long ago.

**Comment by RobFog** [comment]
Shouldn't there be some indication in form of a label or something to indicate this pull request is waiting on a reply to the issue mentioned above?

**Comment by Jojo-Schmitz** [comment]
Maybe, but this is beyond my control. 
Also I doubt whether there will ever be a reply, a full month passed already :-(

**Comment by RobFog** [comment]
Oh, nevermind, I see Dimitri did just that. Sorry for the noise!

**Comment by Jojo-Schmitz** [comment]
For master this needs a rebase meanwhile, not sure whether it does for 3.x too though, and whether it has a chance to get merged into 3.x.

**Comment by Jojo-Schmitz** [comment]
The issue with Mordent, Inverted Mordent and Short Trill just came up in the [German MuseScore forum](https://musescore.org/de/node/310109), and about a year ago already on Transifex by a Japanese translator, as a commet to the translation of Mordent and Inverted mordent.

**Comment by Jojo-Schmitz** [comment]
> Maybe for scores where we do find an `ornamentMordentInverted` we'd need to fix that somehow (like making it an `ornamentMordent`), but what about the `ornamentMordent`? There's no way to tell whether it is the old one and should get changed to the new `ornamentShortTrill`, or whether it is the new one already and needs to stay as is.

Actually we now have such a method, it is being used in several other places now, like for the chord symbols playback

File: libmscore/articulation.cpp [hunk 1]
```diff
@@ -135,6 +135,14 @@ bool Articulation::readProperties(XmlReader& e)
         if (id == SymId::noSym) {
             id = oldArticulationNames2SymId(s);             // compatibility hack for "old" 3.0 scores
         }
+        if (id == SymId::noSym || s == "ornamentMordentInverted") {   // SMuFL < 1.30
+            id = SymId::ornamentMordent;
+        }
+        if (!programVersion.isEmpty() && programVersion < "3.5.1") {
```
Comment (Jojo-Schmitz) [inline libmscore/articulation.cpp:141]: This might need to get adjusted, depending on when and were this PR gets merged

**Comment by Jojo-Schmitz** [comment]
I believe with that out of the way there's no reason to wait on that SMuFL issue anymore, that "question" label could get removed and this PR merged.

File: libmscore/articulation.cpp [hunk 2]
```diff
@@ -135,6 +135,16 @@ bool Articulation::readProperties(XmlReader& e)
         if (id == SymId::noSym) {
             id = oldArticulationNames2SymId(s);             // compatibility hack for "old" 3.0 scores
         }
+        if (id == SymId::noSym || s == "ornamentMordentInverted") {   // SMuFL < 1.30
+            id = SymId::ornamentMordent;
+        }
+
+        QString programVersion = masterScore()->mscoreVersion();
+        if (!programVersion.isEmpty() && programVersion < "3.6") {
```
Comment (Jojo-Schmitz) [inline libmscore/articulation.cpp:143]: This might need to get adjusted, depending on when and were this PR gets merged

**Comment by Jojo-Schmitz** [comment]
I guess I'll include the latest SMuFL changes too, https://github.com/w3c/smufl/pull/135, https://github.com/w3c/smufl/pull/129 and https://github.com/w3c/smufl/pull/120 (where the last one doesn't affect us at all as far as I can tell)

**Comment by mirabilos** [comment]
@Jojo-Schmitz we need to coordinate the sym.{cpp,h} update, I wrote the script I told you about, but it’ll probably need changes for master.

Do you wish for me to branch off this PR of yours, add a commit on top (like I did in #6742 for 3.x), and then you can pull from my branch and update this PR?

**Comment by Jojo-Schmitz** [comment]
Yes, let's do that (now that I've a) included the latest SMuFL updates and b) fixed my most stupid mistakes)

File: libmscore/articulation.cpp (hunk 2 above)
Comment (Jojo-Schmitz) [reply libmscore/articulation.cpp:143]: As the corresponding PR had been merged into 3,x, this here is correct and can stay as is.

**Comment by mirabilos** [comment]
OK, if you merge https://github.com/Jojo-Schmitz/MuseScore/pull/72 into your `smufl` branch, this PR will (AIUI) get updated with my commit on top of yours, and you can easily review the changes there.

**Comment by Jojo-Schmitz** [comment]
I hope I didn't screw that up?

**Comment by mirabilos** [comment]
$ git diff origin/smufl-ms4..jojo/smufl | wc
          0       0       0

Nope, looks good to me ☺

**Comment by Jojo-Schmitz** [comment]
It doesn't happen often to me to be on this side of the fence ;-)

**Comment by Jojo-Schmitz** [comment]
OK, there's one thing left to do I think, see https://github.com/w3c/smufl/issues/119#issuecomment-716217053, we'd need to deal with the Problem: `ornamentMordentInverted` vs. `ornamentMordent` vs. `ornamentShortTrill`. But we'd need to do this for 3.x (where this change hare has been merged already) and for master, in either case when 'importing' pre-3.6 scores, so I guess, I'd better deal with this in separate PRs, meanwhile this here can get merged.

**Comment by Jojo-Schmitz** [comment]
Oops, actually this is exactly what I did already, somehow I managed to confuse myself ;-)
So all is honky-dory and sorted, see https://github.com/musescore/MuseScore/pull/4883/files#diff-aa3aa491a9912868efe44de537be8f729e376388336aa5d0d7de08b126c66e24R142-R147

**Review by mirabilos — APPROVED** [review]

File: fonttools/smufl2sym.sh [hunk 3]
```diff
@@ -19,18 +19,18 @@ jq -f smufl2sym-cmd-debug.jq \
 } >smufl2sym-out-sym.h-SymId
 
 {
-	echo '    // SMuFL standard symbol IDs {{{'
+	echo '    // SMuFL standard symbol names {{{'
 	jq -f smufl2sym-cmd-symNames.jq \
 	    -r <../fonts/smufl/glyphnames.json
-	echo '    // SMuFL standard symbol IDs }}}'
+	echo '    // SMuFL standard symbol namess }}}'
```
Comment (mirabilos) [inline fonttools/smufl2sym.sh:25]: names**s** (two ‘s’)

**Review by mirabilos — CHANGES_REQUESTED** [review]

**Pull request merged**